src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs

```diff
@@ -0,0 +1,241 @@
+                    {
+                        knownPlatforms.Add(platform, 4); // Default version count is 4
+                    }
+                }
+            }
+
+            static (string platformName, int versionPartsCount) ExtractPlatformAndVersionCount(IMethodSymbol method)
+            {
+                var name = method.Name;
+                if (name.EndsWith(VersionSuffix, StringComparison.Ordinal))
+                {
+                    return (name[2..(name.Length - VersionSuffix.Length)], method.Parameters.Length);
+                }
+
+                return (name[2..], 0);
+            }
+
+            static bool NameAndParametersValid(IMethodSymbol method) =>
+                method.Name.StartsWith(IsPrefix, StringComparison.Ordinal) &&
+                (method.Parameters.Length == 0 || method.Name.EndsWith(VersionSuffix, StringComparison.Ordinal));
+        }
+
+        private static void AnalyzeOperation(IOperation operation, OperationAnalysisContext context, PooledDictionary<string, int> knownPlatforms)
+        {
+            if (operation is IInvocationOperation invocation &&
+                methodNames.Contains(invocation.TargetMethod.Name) &&
+                invocation.Arguments.Length > 0 &&
+                invocation.Arguments[0].Value is { } argument &&
+                argument.ConstantValue.HasValue &&
+                argument.ConstantValue.Value is string platformName &&
+                IsNotKnownPlatform(knownPlatforms, platformName))
+            {
+                context.ReportDiagnostic(argument.Syntax.CreateDiagnostic(UnknownPlatform, platformName));
+            }
+        }
+
+        private static void AnalyzeSymbol(Action<Diagnostic> reportDiagnostic, ISymbol symbol, INamedTypeSymbol supportedAttrbute,
+            INamedTypeSymbol unsupportedAttribute, PooledDictionary<string, int> knownPlatforms, CancellationToken token)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                if (supportedAttrbute.Equals(attribute.AttributeClass.OriginalDefinition, SymbolEqualityComparer.Default) ||
+                    unsupportedAttribute.Equals(attribute.AttributeClass.OriginalDefinition, SymbolEqualityComparer.Default))
+                {
+                    AnalyzeAttribute(reportDiagnostic, attribute, knownPlatforms, token);
+                }
+            }
+        }
+
+        private static void AnalyzeAttribute(Action<Diagnostic> reportDiagnostic, AttributeData attributeData, PooledDictionary<string, int> knownPlatforms, CancellationToken token)
+        {
+            var constructorArguments = attributeData.ConstructorArguments;
+
+            if (constructorArguments.Length == 1)
+            {
+                if (constructorArguments[0].Value is string value)
+                {
+                    AnalyzeStringParameter(reportDiagnostic, attributeData.ApplicationSyntaxReference.GetSyntax(token), knownPlatforms, value);
```

Comment:
`ApplicationSyntaxReference` can be null for attributes inherited from base types that are in metadata. I'd suggest adding `attributeData.ApplicationSyntaxReference != null` check at the start of the method.

Reply:
Good point, i guess the API nullability was not annotated yet, updating